snap/local/files/run_dcgm_exporter.sh

```diff
@@ -4,11 +4,17 @@ set -euo pipefail
 # Build the argument list for the dcgm-exporter command
 args=()
 
+# Check if the nv-hostengine-port option is set.
+nv_hostengine_port="$(snapctl get nv-hostengine-port)"
 # Add the dcgm-exporter-address option if it is set. Default: “:9400”
 dcgm_exporter_address="$(snapctl get dcgm-exporter-address)"
 
 if [ -n "$dcgm_exporter_address" ]; then
     args+=("-a" "$dcgm_exporter_address")
 fi
 
+if [ -n "$nv_hostengine_port" ]; then
+    args+=("-r" "localhost:$nv_hostengine_port")
```

Comment:
It's a little different because with `nv-hostengine` you can only configure the port, so the bind address does not change

Reply:
Wait, if you can only configure the port, why does it have `localhost:` in the argument?

Reply:
it can be a little confusing.
* nv-hostengine talks to the GPU and has separate parameters for configuring the port and address it binds to. We only expose the port setting and leave the address binding at its default (localhost)
* dcgm-exporter talks to nv-hostengine to gather the data it exports. It has an `-r` setting to be told which nv-hostengine to talk to (the two don't necessarily have to be on the same machine), and an `-a` setting to determine the `address:port` it binds to. We expose `-a` but not `-r`.
Does this make sense?

Reply:
Ah got it, thanks.